crates/telemetry/src/config.rs

```diff
@@ -0,0 +1,78 @@
+use envconfig::Envconfig;
+use opentelemetry::KeyValue;
+use opentelemetry_semantic_conventions::resource::SERVICE_VERSION;
+
+/// Provides configuration for the telemetry system.
+///
+/// Largely consists of OpenTelemetry variables defined
+/// [here](https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/#general-sdk-configuration)
+/// but also has some Spin specific configuration.
+#[derive(Envconfig, Debug)]
+pub struct Config {
+    /// Disables OpenTelemetry for all signals (traces and metrics).
+    #[envconfig(from = "OTEL_SDK_DISABLED", default = "false")]
+    pub otel_sdk_disabled: bool,
+
+    /// Value of the `service.name` resource attribute.
+    #[envconfig(from = "OTEL_SERVICE_NAME", default = "spin")]
+    pub otel_service_name: String,
+
+    /// Key-value pairs to be used as resource attributes.
+    #[envconfig(from = "OTEL_RESOURCE_ATTRIBUTES", default = "")]
+    pub otel_resource_attributes: KeyValues,
+
+    /// Default exporter endpoint for both traces and metrics.
+    #[envconfig(from = "OTEL_EXPORTER_OTLP_ENDPOINT")]
+    pub otel_exporter_otlp_endpoint: Option<String>,
+
+    /// Exporter endpoint for traces.
+    #[envconfig(from = "OTEL_EXPORTER_OTLP_TRACES_ENDPOINT")]
+    pub otel_exporter_otlp_traces_endpoint: Option<String>,
+
+    /// Exporter endpoint for metrics.
+    #[envconfig(from = "OTEL_EXPORTER_OTLP_METRICS_ENDPOINT")]
+    pub otel_exporter_otlp_metrics_endpoint: Option<String>,
+
+    /// Determines the verbosity of the OpenTelemetry tracing layer.
+    ///
+    /// This is a Spin specific value. This is what allows us to have different levels of verbosity
+    /// in the fmt layer and the otel layer. The fmt layer respects `RUST_LOG` and the otel layer
+    /// respects `OTEL_TRACING_LEVEL`.
+    #[envconfig(from = "OTEL_TRACING_LEVEL", default = "info")]
```

Comment:
Happy to bikeshed 🚲 🏠 this name.

Reply:
We might want to prefix _this_ one with `SPIN_` since it isn't a standard OTEL env var.

Reply:
I had a similar thought, but remembered we inject into the environment variables at `SPIN_*` so I was concerned I was invading on that namespace. Perhaps it's not that big of a deal to steal the name `OTEL_TRACING_LEVEL` from that namespace.

Reply:
> inject into the environment variables at `SPIN_*`

Not sure what you mean here?

Reply:
Ohhh; the env var provider requires a prefix of `SPIN_VARIABLE_`, not just `SPIN_`. We use lots of vars that start with `SPIN_`.